File: WireGuardClientLibAndroid/app/src/main/java/com/example/wireguardclientlib/WireGuardConfiguration.java
package com.example.wireguardclientlib;

import java.util.List;

public class WireGuardConfiguration {
    public String interfacePrivateKey;
    public String peerPublicKey;
    public String peerEndpoint;
    public List<String> allowedIPs;
    public List<String> dnsServers;

    public WireGuardConfiguration(String interfacePrivateKey, String peerPublicKey, String peerEndpoint, List<String> allowedIPs, List<String> dnsServers) {
        this.interfacePrivateKey = interfacePrivateKey;
        this.peerPublicKey = peerPublicKey;
        this.peerEndpoint = peerEndpoint;
        this.allowedIPs = allowedIPs;
        this.dnsServers = dnsServers;
    }
}

File: WireGuardClientLib/Sources/WireGuardClientLib/WireGuardConfiguration.swift
import Foundation

public struct WireGuardConfiguration {
    public let interfacePrivateKey: String
    public let peerPublicKey: String
    public let peerEndpoint: String
    public let allowedIPs: [String]
    public let dnsServers: [String]?
    public let keepAliveInterval: TimeInterval?

    public init(interfacePrivateKey: String,
                peerPublicKey: String,
                peerEndpoint: String,
                allowedIPs: [String],
                dnsServers: [String]? = nil,
                keepAliveInterval: TimeInterval? = 25) {
        self.interfacePrivateKey = interfacePrivateKey
        self.peerPublicKey = peerPublicKey
        self.peerEndpoint = peerEndpoint
        self.allowedIPs = allowedIPs
        self.dnsServers = dnsServers
        self.keepAliveInterval = keepAliveInterval
    }
}
